contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,356 @@
+           'be found [here](https://dateparser.readthedocs.io/en/latest/#supported-languages).\n'
+           '- The default timezone is **the server\'s**. However, it can be specified at the end of each date, '
+           'in both numerical (`+01:00`) or abbreviated format (`CET`).')
+
+    reply_message(msg, message, client)
+
+def parse_error_message(err, message, client):
+    error = ''
+
+    if err.error_id == 1:
+        error = ('Unknown message format.\n\n'
+                 'Usage examples:\n'
+                 '```\n'
+                 '@gcalendar Meeting with John | 2017/03/14 13:37 | 2017/03/14 15:00:01 | GMT\n'
+                 '@gcalendar Trip to LA | tomorrow\n'
+                 '```\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 2:
+        error = ('Unknown date format.\n\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 3:
+        error = ('Unknown timezone.\n\n'
+                 'Please, use a numerical (`+01:00`) or abbreviated (`CET`) timezone.\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+
+    reply_message(error, message, client)
+
+class GCalendarHandler(object):
+    '''
+    This plugin facilitates creating Google Calendar events.
+
+    Usage:
+        For delimited events:
+            @gcalendar <event_title> | <start_date> | <end_date> | <timezone> (optional)
+        For full-day events:
+            @gcalendar <event_title> | <start_date>
+
+    The "event-title" supports all characters but pipes (|)
+
+    Timezones can be specified in both numerical (+00:00) or abbreviated
+    format (UTC).
+    If no timezone is specified, the server's will be considered as default.
+
+    Right now it only works for the calendar set as "primary" in your account.
+    '''
+    def __init__(self):
+        credentials = get_credentials()
+        http = credentials.authorize(httplib2.Http())
+        self.service = discovery.build('calendar', 'v3', http=http)
+
+    def usage(self):
+        return '''
+            This plugin will allow users to create events
+            in a Google Calendar. Users should preface
+            messages with "@gcalendar".
+
+            Before running this, make sure to register a new
+            project in the Google Developers Console
+            (https://console.developers.google.com/start/api?id=calendar)
+            and download "client secret" as a JSON file,
+            and save it to ~/.gcal_client_secret.json.
+            '''
+
+    def triage_message(self, message, client):
+        original_content = message['content']
+
+        is_gcalendar = original_content.startswith('@gcalendar')
+
+        return is_gcalendar
+
+    def handle_message(self, message, client, state_handler):
+        if message['content'] == '@gcalendar help':
+            help_message(message, client)
```

Comment:
Maybe `send_help_message`? That way it will be clear that this function sends the message.